src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -66,21 +71,6 @@
  *   }
  * }
  *
- * interface CheckInterface
- * {
- *   class MyClass { // OK, nested class in interface is always final
```

Comment:
why we remove this ?

Reply:
Because this is a false statement:

```java
public interface TestClass {
	class NonFinal {}
	final class Final {}
}

class A extends TestClass.NonFinal {} // OK

class B extends TestClass.Final {} //  error: cannot inherit from final Final
```